Civi/Core/SqlTriggers/StaticTriggers.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/*
+ +--------------------------------------------------------------------+
+ | CiviCRM version 4.7                                                |
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC (c) 2004-2017                                |
+ +--------------------------------------------------------------------+
+ | This file is a part of CiviCRM.                                    |
+ |                                                                    |
+ | CiviCRM is free software; you can copy, modify, and distribute it  |
+ | under the terms of the GNU Affero General Public License           |
+ | Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+namespace Civi\Core\SqlTriggers;
+
+/**
+ * Build a set of SQL simple, literal SQL triggers.
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC (c) 2004-2017
+ */
+class StaticTriggers {
```

Comment:
Re: `SqlTriggers` namespace -- Agree that should be singular. Pretty much all namespaces can be equally described as singular or plural, and it's nice to have consistent style.

Re: `StaticTriggers` class -- I don't think the singular/plural issue as a matter of style here -- it's really a semantic difference. `SqlTrigger` would just be one object managing one trigger. But if one object manages multiple triggers, then you'd say `SqlTriggers` or `SqlTriggerSet` or `SqlTriggerList` or `SqlTriggerArray` or `SqlTriggerGroup`. Personally, I prefer for the simplest construction that's accurate.

In this case, this instance of `StaticTriggers` contains three triggers that serve a common goal. (The goal is to update `civicrm_case.modified_date` whenever a related activity is inserted, updated, or deleted.)